docs/enhancements/agent-upgrade.md

```diff
@@ -0,0 +1,189 @@
+---
+title: agent-upgrade
+authors:
+  - "@jhernand"
+creation-date: 2022-06-13
+last-updated: 2022-06-13
+---
+
+# Agent upgrade
+
+## Summary
+
+Usually the time since the agent is started and the cluster is installed is
+short. But in some situations it may be long, even many days, and during that
+long time the service may be upgraded and require a newer version of the agent.
```

Comment:
```suggestion
short. But in some situations it may be long, even many days, and during that
long time the service may be upgraded and may require a newer version of the agent.
```

All new agent versions require a rollout of the service pods (as the agent version is delivered as an environment variable). Not all rollouts of service pods come with a newer agent version.

This should probably be clarified